docs/content/topics/project-developers/secrets.md

```diff
@@ -9,239 +9,157 @@ aliases:
+For passing project secrets into a Job, the best practice is to do so via the
+Job's environment. Brigade inherently treats all Job environment variables as
+sensitive by default, so any instance where their values would otherwise be
+logged will be redacted.
 
-Yes. Helm is there to make your life easier, but you can manage project secrets manually.
-You cannot, however, make manual modifications and then go back to using Helm. Doing so
-may result in lost data.
+Project secrets that are accessed directly in a Job, i.e. outside of its
+environment, *will* be leaked, both in the Job Pod resource on the substrate
+and via `brig` commands that consume this information, e.g.
+`brig event get --id <event id>`.
 
-## Examples
-
----
-
-## Fetching secrets from a remote store
-
-Here is an example where we fetch secrets from a remote store
-([Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/))
-for use by one or more Jobs.
-
-To do so, we first run a Job to authenticate with the remote store, fetch all needed secrets
-and save these secret values to a shared directory.  Then, we run a Job that consumes these values.
-
-Here's what our `brigade.js` file looks like:
+Here is an example of passing a project secret into a Job:
 
 ```javascript
-const { events, Job, Group } = require("brigadier");
-
-const sharedMountPrefix = `/mnt/brigade/share`;
-const secrets = [
-  "foo",
-  "bar"
-]
-
-events.on("exec", (event, project) => {
-  // Create Job to fetch secrets
-  secretfetcher = new Job("secretfetcher", "microsoft/azure-cli:latest");
-  secretfetcher.storage.enabled = true;
-
-  // Login as an Azure Service Principal
-  secretfetcher.tasks.push(
-    `az login --service-principal \
-    -u ${project.secrets.spID} \
-    -p '${project.secrets.spPW}' \
-    --tenant ${project.secrets.spTenant}`)
-
-  // Fetch all secrets from the Azure Key Vault
-  for (i in secrets) {
-    secretfetcher.tasks.push(
-      `az keyvault secret show --vault-name ${project.secrets.keyvault} \
-      -n ${secrets[i]} --query value > ${sharedMountPrefix}/${secrets[i]}`);
-  }
-
-  // Create Job to consume secrets
-  secretconsumer = new Job("secretconsumer", "alpine");
-  secretconsumer.storage.enabled = true;
-
-  // Consume all secrets
-  for (i in secrets) {
-    secretconsumer.tasks.push(
-      `ls -haltr ${sharedMountPrefix}/${secrets[i]}`);
-  }
-
-  // Run jobs sequentially
-  Group.runEach([secretfetcher, secretconsumer]);
-})
-```
+const { Job, events } = require("@brigadecore/brigadier");
 
-To run this example, we replace `BRIGADE_PROJECT_NAME` with our Brigade project name and run:
+events.on("brigade.sh/cli", "exec", async event => {
+  let job = new Job("second-job", "debian:latest", event);
+  job.primaryContainer.environment = {"FOO_SECRET": event.project.secrets.foo};
+  job.primaryContainer.command = ["bash"];
+  job.primaryContainer.arguments = ["-c", "echo $FOO_SECRET | wc -c"];
```

Comment:
This example has an interesting mix of good and bad practices. It's good that the secret isn't included directly in the command where it can be seen by anyone who can view the pod. The command gets the secret from an env var. This is a _great_ pattern, I think. But... as far as that the command _does_... it's putting the secret straight into the logs, which is definitely a problem.  Is there something interesting we could do with the secret without printing it?

Reply:
Good point.  Updated with a docker login example.  WDYT?